mono/metadata/reflection.c

```diff
@@ -1549,7 +1549,7 @@ assembly_name_to_aname (MonoAssemblyName *assembly, char *p) {
 					p++;
 				}
 			}
-		} else if (*p == 'P' && g_ascii_strncasecmp (p, "PublicKeyToken=", 15) == 0) {
+		} else if ((*p == 'P' || *p == 'p') && g_ascii_strncasecmp (p, "PublicKeyToken=", 15) == 0) {
```

Comment:
Please consider a macro:
#define G_STRING_CONSTANT_AND_LENGTH(x) (x), sizeof (x) - 1
to replace `8` and `15`.